src/main/java/com/capitalone/dashboard/collector/GitHubCollectorTask.java

```diff
@@ -169,10 +169,11 @@ public void collect(Collector collector) {
         LOG.info("GitHubCollectorTask:collect start, total enabledRepos=" + enabledRepos.size());
         for (GitHubRepo repo : enabledRepos) {
             repoCount++;
-            String repoUrl = repo.getRepoUrl() + "/tree/" + repo.getBranch();
+            String repoUrl = repo==null?"null":(repo.getRepoUrl() + "/tree/" + repo.getBranch());
             String statusString = "UNKNOWN";
-            long lastUpdated = repo.getLastUpdated();
+            long lastUpdated = repo==null?0:repo.getLastUpdated();
             try {
+                if (repo==null) throw new HygieiaException("Repository returned from github is null", HygieiaException.BAD_DATA);
```

Comment:
shouldn't the null check be the first line inside the for loop? Also if the repo is null we need to log an error and continue to the next repo.

Reply:
I would like to have the same log line in "finally" to log for every case, in order for splunk to report all data as well as statistics, that is why I do not use a different log line.